xseed.sh

```diff
@@ -43,22 +43,65 @@ else
 fi
 
 # Use environment variables with descriptive default values
-TORRENT_CLIENT_NAME=${TORRENT_CLIENT_NAME:-Qbit}
-USENET_CLIENT_NAME=${USENET_CLIENT_NAME:-SABnzbd}
+# For multiple clients, use format: client1,client2,client3
+TORRENT_CLIENTS=${TORRENT_CLIENTS:-"Qbit"}
+USENET_CLIENTS=${USENET_CLIENTS:-"SABnzbd"}
 XSEED_HOST=${XSEED_HOST:-crossseed}
 XSEED_PORT=${XSEED_PORT:-8080}
 LOG_FILE=${LOG_FILE:-/config/xseed.log}
 LOGID_FILE=${LOGID_FILE:-/config/xseed-id.log}
 XSEED_APIKEY=${XSEED_APIKEY:-}
+
+# Save original IFS
+OLD_IFS="$IFS"
+
+# Convert comma-separated strings to arrays
+IFS=',' read -r -a TORRENT_CLIENT_ARRAY <<< "$TORRENT_CLIENTS"
+IFS=',' read -r -a USENET_CLIENT_ARRAY <<< "$USENET_CLIENTS"
+
+# Restore original IFS
+IFS="$OLD_IFS"
+
+# Trim whitespace from array elements
+TORRENT_CLIENT_ARRAY=("${TORRENT_CLIENT_ARRAY[@]/#[[:space:]]/}")
+TORRENT_CLIENT_ARRAY=("${TORRENT_CLIENT_ARRAY[@]/%[[:space:]]/}")
+USENET_CLIENT_ARRAY=("${USENET_CLIENT_ARRAY[@]/#[[:space:]]/}")
+USENET_CLIENT_ARRAY=("${USENET_CLIENT_ARRAY[@]/%[[:space:]]/}")
+
 log_message "DEBUG" "Using '.env' file for config?: $EVAR"
-log_message "INFO" "Using Configuration:"
-log_message "INFO" "TORRENT_CLIENT_NAME=$TORRENT_CLIENT_NAME"
-log_message "INFO" "USENET_CLIENT_NAME=$USENET_CLIENT_NAME"
+log_message "INFO" "Using configuration:"
+log_message "INFO" "TORRENT_CLIENTS=$TORRENT_CLIENTS"
+log_message "INFO" "USENET_CLIENTS=$USENET_CLIENTS"
 log_message "INFO" "XSEED_HOST=$XSEED_HOST"
 log_message "INFO" "XSEED_PORT=$XSEED_PORT"
 log_message "INFO" "LOG_FILE=$LOG_FILE"
 log_message "INFO" "LOGID_FILE=$LOGID_FILE"
 
+# Function to check if a client is in the allowed list
+is_valid_client() {
+    local client="$1"
+    local client_type="$2"
+    local valid=false
+    
+    if [ "$client_type" == "torrent" ]; then
+        for allowed_client in "${TORRENT_CLIENT_ARRAY[@]}"; do
+            if [ "$client" == "$allowed_client" ]; then
+                valid=true
+                break
+            fi
+        done
+    elif [ "$client_type" == "usenet" ]; then
+        for allowed_client in "${USENET_CLIENT_ARRAY[@]}"; do
+            if [ "$client" == "$allowed_client" ]; then
+                valid=true
+                break
+            fi
+        done
+    fi
+    
+    echo "$valid"
+}
```

Comment:
```suggestion
is_valid_client() {
    local client="$1"
    local client_type="$2"

    case $client_type in
        "torrent")
            for allowed_client in "${TORRENT_CLIENTS[@]}"; do
                if [[ "$client" == "$allowed_client" ]]; then
                    return 0
                    break
                fi
            done
            ;;
        "usenet")
            for allowed_client in "${USENET_CLIENTS[@]}"; do
                if [[ "$client" == "$allowed_client" ]]; then
                    return 0
                    break
                fi
            done
            ;;
    esac

    return 1
}
```

Refactored and made it cleaner with a case for the client_type, and returned in the case if found rather than relying on a secondary variable just to return later.

Also covered what I said in the other review...

> We don't need to echo when we can simply return, as this echo is only used as a boolean value, return 0 is success, 1 is failure.